platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/DTCoreText/DTHTMLElement.swift

```diff
@@ -35,4 +35,70 @@ extension DTHTMLElement {
             }
         }
     }
+
+    func clearTrailingAndLeadingNewlinesInCodeblocks() {
+        guard let childNodes = childNodes as? [DTHTMLElement] else {
+            return
+        }
+
+        if name == "pre",
+           childNodes.count == 1,
+           let child = childNodes.first as? DTTextHTMLElement,
+           var text = child.text(),
+           text != .nbsp {
+            var leadingDiscardableElement: DiscardableTextHTMLElement?
+            var trailingDiscardableElement: DiscardableTextHTMLElement?
+            var shouldReplaceNodes = false
+
+            if text.hasPrefix("\(Character.nbsp)") {
+                shouldReplaceNodes = true
+                text.removeFirst()
+                leadingDiscardableElement = createDiscardableElement()
+            }
+            
+            if text.hasSuffix("\(Character.nbsp)") {
+                shouldReplaceNodes = true
+                text.removeLast()
+                trailingDiscardableElement = createDiscardableElement()
+            }
+
+            if shouldReplaceNodes {
+                removeAllChildNodes()
+
+                if let leadingDiscardableElement = leadingDiscardableElement {
+                    addChildNode(leadingDiscardableElement)
+                    addChildNode(createLineBreak())
+                }
+
+                let newTextNode = DTTextHTMLElement()
+                newTextNode.inheritAttributes(from: self)
+                newTextNode.interpretAttributes()
+                newTextNode.setText(text)
+                addChildNode(newTextNode)
```

Comment:
I think you might be able to re-use the same text node after detaching it instead of creating a new one (no need to re-inherit/interpret attributes then)